# Config file for Joy Manager
#############################

# Joysticks that are in use. Priority has to differ for all of them:
joysticks:
  - name: rqt
    topic: /anyjoy/rqt
    priority: 1
  - name: operator
    topic: /anyjoy/operator
    priority: 2
  - name: onboard
    topic: /anyjoy/onboard
    priority: 3

# Output publishers that are in use.
outputs:
  - name: joy_output
    topic: /commands/joy
    type: Joy
    is_publishing: true
  - name: twist_output
    topic: /commands/twist
    type: Twist
    is_publishing: true
  - name: pose_output
    topic: /commands/pose
    type: Pose
    is_publishing: true

# Timer: Seconds that pass before the priority is reset for all incomming messages
timer: 5.0

# Timeout: Seconds that pass before the last received message is no longer repeated
timeout: 10.0

# Timestep: Any_node time step inbetween calls to update(const any_worker::WorkerEvent& event)
time_step: 0.1

standalone: true

publishers:
  soft_emergency:
    topic: /soft_emcy_stop
    queue_size: 10
    latch: false
  hard_emergency:
    topic: /hard_emcy_stop
    queue_size: 10
    latch: false

notification:
  notifiers:
    default:
      topic: /notification
      queue_size: 100
      latch: false
      output_devices:
        - "onboard_screen"
        - "joystick_screen"
        - "operator_screen"

# Default values for the mapping
# of joystick axes [-1,1] to
# twist [twistMin, twistMax] and
# pose [poseMin, poseMax]:
twist_output:
  frame_id: base
  subscribers:
    twist_min:
      topic: /commands/twist_min
      queue_size: 10
    twist_max:
      topic: /commands/twist_max
      queue_size: 10
  twistMin:
    linear:
      x: -1.0
      y: -1.0
      z: -1.0
    angular:
      x: -1.0
      y: -1.0
      z: -1.0
  twistMax:
    linear:
      x: 1.0
      y: 1.0
      z: 1.0
    angular:
      x: 1.0
      y: 1.0
      z: 1.0

pose_output:
  frame_id: base
  subscribers:
    pose_min:
      topic: /commands/pose_min
      queue_size: 10
    pose_max:
      topic: /commands/pose_max
      queue_size: 10
  poseMin:
    position:
      x: -1.0
      y: -1.0
      z: -1.0
    orientation:
      x: -0.462
      y: -0.191
      z: -0.462
      w: 0.733
  poseMax:
    position:
      x: 1.0
      y: 1.0
      z: 1.0
    orientation:
      x: 0.191
      y: 0.462
      z: 0.191
      w: 0.845
